lib/eof_test.go

```diff
@@ -0,0 +1,38 @@
+package lib
+
+import (
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/hashicorp/yamux"
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrIsEOF(t *testing.T) {
+	t.Parallel()
```

Comment:
I know a lot of the tests use `t.Parallel`, but it can cause problems sometimes so I've been trying to remove it where possible. I think in this case, since these tests run so fast, it's actually slower to spin up the goroutine to run the tests in parallel, than it would be to run it without parallel.

```suggestion
```